controllers/train.py

```diff
@@ -0,0 +1,96 @@
+from bson.json_util import dumps
+from flask import Blueprint, make_response, request
+import requests
+import json
+import datetime
+from cerberus import Validator
+
+# create the blueprint (controller) for the trains
+trains = Blueprint('trains', __name__)
+
+# schema validator
+def schema_validator(arrival=None, departure=None, departure_date=None):
+    schema = {'arrival':{'type':'string', 'maxlength':3}, 'departure':{'type':'string','maxlength':3}, 'departure_date':{'type':'string', 'maxlength':10}}
+    v = Validator(schema)
+    input_info = {}
+    # the proper string pair with its variable
+    all_vali = [('arrival', arrival), ('departure', departure), ('departure_date', departure_date)]
+    # iterate over all passed variable
+    for info in all_vali:
+        # unpack tuple
+        info_key, info_value = info
+        # if the variable is not 'None' pass it through schema validator
+        if info_value:
+            input_info[info_key] = info_value
+    return v.validate(input_info)
+
+# getting trip info from each train
+def trip_info(trip):
+    # variable to store the info we actually want
+    trip_data = {}
+    # array with the keys we want
+    train_keys = ['origin','destination','train','consistNumber','estimatedDeparture',
+               'scheduledDeparture','actualDeparture','scheduledArrival','estimatedArrival',
+               'actualArrival','actualArrivalPlatform','actualDeparturePlatform',
+               'arrivalPlatform','arrivalStatus','departurePlatform','legs']
+    # array with the keys we want from 'leg'
+    leg_keys = ['actualArrival','actualDeparture','estimatedArrival','estimatedDeparture',
+                'arrivalStatus','consistNumber','isCancelled','actualArrivalPlatform',
+                'actualDeparturePlatform','arrivalPlatform','departurePlatform','legStatus',
+                'origin','scheduledArrival','scheduledArrivalPlatform','scheduledDeparture',
+                'scheduledDeparturePlatform','train']
+    for key in train_keys:
+        # check if key is missing
+        try:
+            # get value of the current key
+            trip_value = trip[key]
+            if key == 'legs':
+                # array to hold all 'leg' dictionaries with keys we want
+                leg_data = []
+                for leg in trip_value:
+                    # variable to store the info we want from 'leg'
+                    leg_info = {}
+                    for leg_key in leg_keys:
+                        leg_info[leg_key] = leg[leg_key]
+                    leg_data.append(leg_info)
+                trip_data[key] = leg_data
+                continue
```

Comment:
Unnecessary continue

Reply:
without the continue then trip_data[key] will be assigned to the unclean leg value, which has keys we don't need.

Reply:
Didn't see that the try catch was in the loop, and using the exception instead of a check of string length is pretty pythonic, nice method.